components/ai_chat/resources/page/components/main/index.tsx

```diff
@@ -67,52 +79,69 @@ function Main() {
 
   let currentErrorElement = null
 
-  let scrollerElement: HTMLDivElement | null = null
   const headerElement = React.useRef<HTMLDivElement>(null)
   const conversationContentElement = React.useRef<HTMLDivElement>(null)
 
-  const scrollPos = React.useRef({ isAtBottom: true })
-
   // Determine which, if any, error message should be displayed
   if (aiChatContext.hasAcceptedAgreement && conversationContext.apiHasError) {
     switch (conversationContext.currentError) {
-      case mojom.APIError.ConnectionIssue:
+      case Mojom.APIError.ConnectionIssue:
         currentErrorElement = <ErrorConnection
           onRetry={conversationContext.retryAPIRequest} />
         break
-      case mojom.APIError.RateLimitReached:
+      case Mojom.APIError.RateLimitReached:
         currentErrorElement = <ErrorRateLimit />
         break
-      case mojom.APIError.ContextLimitReached:
+      case Mojom.APIError.ContextLimitReached:
         currentErrorElement = <ErrorConversationEnd />
         break
-      case mojom.APIError.InvalidEndpointURL:
+      case Mojom.APIError.InvalidEndpointURL:
         currentErrorElement = <ErrorInvalidEndpointURL />
         break
     }
   }
 
-  const handleScroll = (e: React.UIEvent<HTMLDivElement>) => {
-    // Monitor scroll positions only when Assistant is generating
-    if (!conversationContext.isGenerating) return
-    const el = e.currentTarget
-    scrollPos.current.isAtBottom = Math.abs(el.scrollHeight - el.clientHeight - el.scrollTop) < SCROLL_BOTTOM_THRESHOLD
-  }
+  // Automatic scroll to bottom of scroll anchor when generating new response lines
+  const scrollIsAtBottom = React.useRef(true)
+  const scrollElement = React.useRef<HTMLDivElement | null>(null)
+  const scrollAnchor = React.useRef<HTMLDivElement | null>(null)
 
-  const handleLastElementHeightChange = () => {
-    if (!scrollerElement) {
-      return
+  const handleScroll = React.useCallback((e: React.UIEvent<HTMLDivElement>) => {
+    // Monitor scroll positions only when Assistant is generating,
+    // but always reset to bottom before next generation
+    if (!conversationContext.isGenerating) {
```

Comment:
Because when you submit the next entry I (we?) want to scroll down to the bottom again. It will only use the value of `scrollIsAtBottom` when it's generating.